provider/consumer.py

```diff
@@ -291,6 +294,12 @@ def __pollForMessages(self):
         self.updateLastPoll()
         return messages
 
+    # decide whether or not to disable a trigger based on the status code returned
+    # from firing the trigger. Specifically, disable on all 4xx status codes
+    # except 408 (gateway timeout) and 429 (throttle)
+    def __shouldDisable(self, status_code):
+        return status_code in range(400, 500) and status_code not in [408, 429]
```

Comment:
This means, we disable for Not Found (404), Unauthorized (401) or Forbidden (403) which is think is correct, this can be clean at a later time by another external process.